### ISSUE TITLE
New share dialogue 

### ISSUE THREAD

**Issue by TecJon** [nextcloud/ios#764]
I'd like to propose a more intuitive share dialogue. The current share view is rather complicated and it can be confusing at times. A client recently failed to share a file because she did not understand the UI. Ideally, the two views (general share settings & find user/group) can be combined to one simpler view.

### Combining search for a username and enter username
Currently, there are two input fields for entering a known username and searching for a username. This is really confusing. The search doesn't say anything if nothing was found. There are long text labels explaining a simple operation.

The current screen to select a user/group: 
<img src="https://user-images.githubusercontent.com/11375752/50852470-e24d4580-137f-11e9-837f-119d4dfb9343.PNG" width="250">

I think this can be reduced to a **single input field**. Dropbox has solved the share screen very elegantly:

<img src="https://user-images.githubusercontent.com/11375752/50852626-60115100-1380-11e9-8bde-d8afcb09413b.PNG" width="450">

Similar to dropbox I suggest for NC a single input field. 
- Typing in a valid group/user name followed by "Return" will add the share.
- If the user didn't type a valid username or group name the search result is shown below the input field. 
- If the user typed an email address the (federated) email share option could be triggered (like on the web interface) 

Each added share with a user or group appears below the input field. The username or group name is prepended with an group icon for groups and the avatar for users. Swiping left on the share removes it. 
Using a single input field with a meaningful placeholder (like dropbox) will allow to remove all of the explanatory text. 
**The reduced size allows the input field to be integrated in the first share view. Combining the two views into one.**

### Simplified look
It could look something like this:

![newshare2](https://user-images.githubusercontent.com/11375752/50863363-b1c8d400-139e-11e9-9dfd-6ea7edc12371.PNG)

Tapping on "create a link", a pop up appears (taken from dropbox):

<img src="https://user-images.githubusercontent.com/11375752/50864274-c9ee2280-13a1-11e9-813d-92146d61d1b5.PNG" width="320">

Two users added and link sharing:

![newshare2a](https://user-images.githubusercontent.com/11375752/50864985-0fabea80-13a4-11e9-9d0e-67c766f94ce7.PNG)

What do you think?

**Comment by marinofaggiana** [comment]
Hi @TecJon thanks for your great job, soon I study your scheme, the remake of share dialogue is in ToDo ;)

**Comment by Peterkrijgsman85** [comment]
This is a really good replacement for the current share sheet. 
Personaly I want to have the email option in the app. This is now only available in the web interface.

**Comment by mhls1981** [comment]
> Personaly I want to have the email option in the app. This is now only available in the web interface.

I agree - I really do not get why the share via email does not work through the app. Would be awesome (and for most users using it considered "standard").
